### DISCUSSION TITLE
Adding Dynamic routes during runtime

### DISCUSSION THREAD

**Discussion by VaZark** [tokio-rs/axum#710] [Ideas]
Is there any way to add and remove routes dynamically at runtime?

**Usecase:** When building a plugin-based application, it'd be interesting to allow plugins to register/add routes at runtime and deregister/delete on disabling or removing the plugin.

I had posed the question a few weeks ago and got some help already. Here is the snippet that allows dynamic addition of GET routes : [gist](https://gist.github.com/davidpdrsn/2890612d4e2b9e419b3bf64fce081272). The added code doesn't actually run but the route does get added though.

Can this be exposed as an API? Something like `NestedDynamicRouter` ?

**Comment by davidpdrsn** [comment]
> The added code doesn't actually run but the route does get added though.

I don't understand what you mean. Can you elaborate?

> Can this be exposed as an API? Something like NestedDynamicRouter ?

No I don't think we're gonna build this into axum directly. Its quite a niche use case and possible today without any new API in axum.

**Comment by norman784** [comment]
I'd might be wrong but maybe reverse proxy is what you are looking for.
